router/src/lib.rs

```diff
@@ -1002,21 +996,18 @@ pub struct FunctionName {
 
 #[derive(Debug, Clone, PartialEq, Serialize, Deserialize, ToSchema, Default)]
 #[serde(from = "ToolTypeDeserializer")]
+#[serde(rename_all = "snake_case")]
 /// <https://platform.openai.com/docs/guides/function-calling/configuring-function-calling-behavior-using-the-tool_choice-parameter>
-pub enum ChatCompletionToolChoiceOption {
+pub enum ToolChoice {
     /// Means the model can pick between generating a message or calling one or more tools.
-    #[schema(rename = "auto")]
+    #[default]
     Auto,
     /// Means the model will not call any tool and instead generates a message.
     #[schema(rename = "none")]
-    #[default]
     NoTool,
```

Comment:
Isn't this `no_tool` with `snake_case`  ?This should mean a rename of this property or `None`,  no ?

I don't think `schema(rename)` imples a serde `rename`.

Reply:
good catch, updated to `serde(rename)` and added a test to ensure that the `"none"` values is respected correctly